packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +189,318 @@ class Xcode {
     );
   }
 }
+
+/// A utility class for interacting with Xcode xcdevice command line tools.
+class XCDevice {
+  XCDevice({
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required Xcode xcode,
+  }) : _processUtils = ProcessUtils(logger: logger, processManager: processManager),
+       _logger = logger,
+       _xcode = xcode;
+
+  final ProcessUtils _processUtils;
+  final Logger _logger;
+  final Xcode _xcode;
+
+  bool get isInstalled => _xcode.isInstalledAndMeetsVersionCheck && xcdevicePath != null;
+
+  String _xcdevicePath;
+  String get xcdevicePath {
+    if (_xcdevicePath == null) {
+      try {
+        _xcdevicePath = _processUtils.runSync(
+          <String>[
+            'xcrun',
+            '--find',
+            'xcdevice'
+          ],
+          throwOnError: true,
+        ).stdout.trim();
+      } on ProcessException catch (exception) {
+        _logger.printTrace('Process exception finding xcdevice:\n$exception');
+      } on ArgumentError catch (exception) {
+        _logger.printTrace('Argument exception finding xcdevice:\n$exception');
+      }
+    }
+    return _xcdevicePath;
+  }
+
+  Future<List<dynamic>> _getAllDevices({bool useCache = false}) async {
+    if (!isInstalled) {
+      _logger.printTrace('Xcode not found. Run \'flutter doctor\' for more information.');
+      return null;
+    }
+    if (useCache && _cachedListResults != null) {
+      return _cachedListResults;
+    }
+    try {
+      // USB-tethered devices should be found quickly. 1 second timeout is faster than the default.
+      final RunResult result = await _processUtils.run(
+        <String>[
+          'xcrun',
+          'xcdevice',
+          'list',
+          '--timeout',
+          '1',
+        ],
+        throwOnError: true,
+      );
+      if (result.exitCode == 0) {
+        final List<dynamic> listResults = json.decode(result.stdout) as List<dynamic>;
+        _cachedListResults = listResults;
+        return listResults;
+      }
+      _logger.printTrace('xcdevice returned an error:\n${result.stderr}');
+    } on ProcessException catch (exception) {
+      _logger.printTrace('Process exception running xcdevice list:\n$exception');
+    } on ArgumentError catch (exception) {
+      _logger.printTrace('Argument exception running xcdevice list:\n$exception');
+    } on FormatException catch (exception) {
+      // Failed to parse the xcdevice output, or it returned junk.
+      _logger.printError('xcdevice returned non-JSON response:\n$exception');
```

Comment:
might this happen legitimately for reasons other than the format was changed? i'm wondering if this should be an exception and so we can surface this.

Reply:
Yeah that makes sense, we should see the crashes if this happens.